File: milventory/src/constants/mapBounds.js
/** Room bounds in world/map coordinates — must match src/api/helpers/map_bounds.py */
export const MAP_ROOM = {
  x: 80,
  y: 80,
  width: 3600,
  height: 3840
};

export function clampPointToRoom(x, y) {
  const maxX = MAP_ROOM.x + MAP_ROOM.width;
  const maxY = MAP_ROOM.y + MAP_ROOM.height;
  return {
    x: Math.round(Math.max(MAP_ROOM.x, Math.min(maxX, x))),
    y: Math.round(Math.max(MAP_ROOM.y, Math.min(maxY, y)))
  };
}

export function pointInRoom(x, y) {
  const p = clampPointToRoom(x, y);
  return p.x === Math.round(x) && p.y === Math.round(y);
}

File: src/api/helpers/map_bounds.py
"""
Map placement bounds (world coordinates). Must match frontend inventoryBounds.room defaults.
"""
# Same as milventory InventoryContext default room rect
ROOM_X = 80
ROOM_Y = 80
ROOM_WIDTH = 3600
ROOM_HEIGHT = 3840


def coords_in_room(x, y):
    """True if integer coords lie inside the closed room rectangle."""
    try:
        xi = int(round(float(x)))
        yi = int(round(float(y)))
    except (TypeError, ValueError):
        return False
    if xi < ROOM_X or yi < ROOM_Y:
        return False
    if xi > ROOM_X + ROOM_WIDTH or yi > ROOM_Y + ROOM_HEIGHT:
        return False
    return True


def clamp_coords_to_room(x, y):
    """Return (int, int) clamped to room bounds."""
    try:
        xf = float(x)
        yf = float(y)
    except (TypeError, ValueError):
        xf = float(ROOM_X)
        yf = float(ROOM_Y)
    xi = int(round(max(ROOM_X, min(ROOM_X + ROOM_WIDTH, xf))))
    yi = int(round(max(ROOM_Y, min(ROOM_Y + ROOM_HEIGHT, yf))))
    return xi, yi
